server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -236,6 +235,16 @@ public int add(InputRow row, Supplier<Committer> committerSupplier) throws Index
       persist(committerSupplier.get());
     }
 
+    if (++counter % CHECK_INTERNAL_COUNT == 0) {
+      int size = 0;
+      for (Sink aSink : sinks.values()) {
+        size += aSink.size();
+      }
+      if (size > config.getMaxRowsInMemory()) {
+        persistOldest(committerSupplier.get());
```

Comment:
since this persist is triggered by the maxRows config, would it make sense to persist the one having the maxSize ? as the oldest one may have very small number of rows.